### PR TITLE
GSOC-22:Ver 1-S3 Upload Script

### PR DESCRIPTION
**Pull request by ag-cdev**
Status:

S3 Upload is Working only need one params that is passed through the shell script.

Next Task:

Looking to enable renamer of files to better organize it.
Slowly remove the shell script as a whole after other cloud's like dropbox, gdrive are integrated.

### CODE REVIEW

**Comment by ag-cdev** [comment]
Uploads Working:
S3
Dropbox
Box

File: magic_script.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+import logging
+import boto3
+from sys import argv
+from botocore.exceptions import ClientError
+import sys
+import dropbox
+from boxsdk import OAuth2, Client
+
+
+CONFIGFILE = "/mnt/gsoc/test/s3/config.json"
```
Comment (aaronkvanmeerten) [inline magic_script.py]: Please do not hardcode paths like this in your scripts.  This should assume a relative path of the present directory.

File: magic_script.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
+import json
```
Comment (aaronkvanmeerten) [inline magic_script.py]: Please rename this script to python_uploader.py

File: config.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (aaronkvanmeerten) [inline config.json]: Please rename this to config.json.example to avoid actually having it parsed by your code, since its' not actually safe for everyone.

**Review by aaronkvanmeerten — CHANGES_REQUESTED** [review]
most of this looks like good, a few comments and we can merge it.

**Comment by ag-cdev** [comment]
Please check all changes done @aaronkvanmeerten

**Comment by palmtown** [comment]
> Please check all changes done @aaronkvanmeerten

The status of this PR appears to be still "1 review requesting changes."  Can you push it to review?  Also, how to install and use this with s3?

**Comment by aaronkvanmeerten** [comment]
This was a gsoc project which has not had any attention since last year, I am closing it for now but leaving the branch in case anyone is interested in the implementation.

**Comment by aaronkvanmeerten** [comment]
> > Please check all changes done @aaronkvanmeerten
> 
> The status of this PR appears to be still "1 review requesting changes." Can you push it to review? Also, how to install and use this with s3?

Sorry to say I'm not sure how to configure it with s3 and am lacking the available time to document or support it.  Feel free to review the code on the branch and see if it's of use to you, and you could install it over the existing .deb installer version.

**Pull request closed without merging**